Unity/Package/Tests/Runtime/V8HostObjectTest.cs

```diff
@@ -0,0 +1,436 @@
+﻿using System;
+using System.Globalization;
+using Microsoft.ClearScript.V8;
+using Microsoft.ClearScript.V8.SplitProxy;
+using NUnit.Framework;
+
+namespace Microsoft.ClearScript.Test
+{
+    [TestFixture, Parallelizable(ParallelScope.Fixtures)]
+    internal sealed class V8HostObjectTest
+    {
+        private V8ScriptEngine engine;
+        private HostObject hostObject;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            engine = new V8ScriptEngine(V8ScriptEngineFlags.EnableDateTimeConversion);
+            hostObject = new HostObject();
+            engine.AddHostObject("hostObject", hostObject);
+
+            engine.AddHostObject("assert", new InvokeHostObject((args, result) =>
+            {
+                if (args.Length != 1)
+                    throw new ArgumentException($"Expected 1 argument, but got {args.Length}");
+                
+                Assert.That(args[0].GetBoolean());
+            }));
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            engine.Dispose();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            hostObject.GetNamedProperty = null;
+            hostObject.SetNamedProperty = null;
+            hostObject.DeleteNamedProperty = null;
+            hostObject.GetIndexedProperty = null;
+            hostObject.SetIndexedProperty = null;
+            hostObject.DeleteIndexedProperty = null;
+            hostObject.GetEnumertor = null;
+            hostObject.GetAsyncEnumerator = null;
+            hostObject.GetNamedPropertyNames = null;
+            hostObject.GetIndexedPropertyIndices = null;
+        }
+
+        [Test]
+        public void DeleteIndexedProperty()
+        {
+            bool wasCalled = false;
+
+            hostObject.DeleteIndexedProperty = index =>
+            {
+                wasCalled = true;
+                Assert.That(index == 42);
+                return true;
+            };
+            
+            engine.Execute(@"{
+                delete hostObject[42];
+            }");
+            
+            Assert.That(wasCalled);
+        }
+        
+        [Test]
+        public void DeleteNamedProperty()
+        {
+            bool wasCalled = false;
+            
+            hostObject.DeleteNamedProperty = name =>
+            {
+                wasCalled = true;
```

Comment:
It's not usually how you do it:

Consider the following instead if usage of `interface` or `abstract` is possible instead of the final `HostObject`:
- Mock via `NSubstitute`
- Invoke `<FunctionName>.Recieved(<check pattern goes here>)`

Reply:
Substitute can't deal with `readonly ref struct` types because they cannot be used as generic arguments.